anomvalidate/src/main/java/com/expedia/adaptivealerting/anomvalidate/filter/InvestigationFilter.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2018 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.anomvalidate.filter;
+
+import com.expedia.adaptivealerting.core.anomaly.AnomalyResult;
+
+public interface InvestigationFilter {
+    boolean keep(String key, AnomalyResult anomalyResult);
```

Comment:
Can you Javadoc this class and method? Not sure what these are exactly.

Reply:
Oh, I see. This is an abstraction over the pre- and post-filter implementations.